src/mlpack/tests/convolutional_network_test.cpp

```diff
@@ -129,4 +129,79 @@ BOOST_AUTO_TEST_CASE(VanillaNetworkTest)
   BOOST_REQUIRE_EQUAL(success, true);
 }
 
+/**
+ * Test that FFN::Train() returns finite final objective
+ * value.
+ */
+BOOST_AUTO_TEST_CASE(CNNTrainReturnObjective)
+{
+  arma::mat X;
+  X.load("mnist_first250_training_4s_and_9s.arm");
+
+  // Normalize each point since these are images.
+  arma::uword nPoints = X.n_cols;
+  for (arma::uword i = 0; i < nPoints; i++)
+  {
+    X.col(i) /= norm(X.col(i), 2);
+  }
+
+  // Build the target matrix.
+  arma::mat Y = arma::zeros<arma::mat>(1, nPoints);
+  for (size_t i = 0; i < nPoints; i++)
+  {
+    if (i < nPoints / 2)
+    {
+      // Assign label "1" to all samples with digit = 4
+      Y(i) = 1;
+    }
+    else
+    {
+      // Assign label "2" to all samples with digit = 9
+      Y(i) = 2;
+    }
+  }
+
+  /*
+   * Construct a convolutional neural network with a 28x28x1 input layer,
+   * 24x24x8 convolution layer, 12x12x8 pooling layer, 8x8x12 convolution layer
+   * and a 4x4x12 pooling layer which is fully connected with the output layer.
+   * The network structure looks like:
+   *
+   * Input    Convolution  Pooling      Convolution  Pooling      Output
+   * Layer    Layer        Layer        Layer        Layer        Layer
+   *
+   *          +---+        +---+        +---+        +---+
+   *          | +---+      | +---+      | +---+      | +---+
+   * +---+    | | +---+    | | +---+    | | +---+    | | +---+    +---+
+   * |   |    | | |   |    | | |   |    | | |   |    | | |   |    |   |
+   * |   +--> +-+ |   +--> +-+ |   +--> +-+ |   +--> +-+ |   +--> |   |
+   * |   |      +-+   |      +-+   |      +-+   |      +-+   |    |   |
+   * +---+        +---+        +---+        +---+        +---+    +---+
+   */
+
+
+  FFN<NegativeLogLikelihood<>, RandomInitialization> model;
+
+  model.Add<Convolution<> >(1, 8, 5, 5, 1, 1, 0, 0, 28, 28);
+  model.Add<ReLULayer<> >();
+  model.Add<MaxPooling<> >(8, 8, 2, 2);
+  model.Add<Convolution<> >(8, 12, 2, 2);
+  model.Add<ReLULayer<> >();
+  model.Add<MaxPooling<> >(2, 2, 2, 2);
+  model.Add<Linear<> >(192, 20);
+  model.Add<ReLULayer<> >();
+  model.Add<Linear<> >(20, 10);
+  model.Add<ReLULayer<> >();
+  model.Add<Linear<> >(10, 2);
+  model.Add<LogSoftMax<> >();
+
+  // Train for only 8 epochs.
+  ens::RMSProp opt(0.001, 1, 0.88, 1e-8, 8 * nPoints, -1);
+
+  double objVal = model.Train(X, Y, opt);
+
+  // Test that objective value returned by FFN::Train() is finite.
+  BOOST_REQUIRE_EQUAL(std::isfinite(objVal), true);
```

Comment:
I wonder if it makes sense to include the check into the existing cnn test just to keep the test time low. Let me know what you think.

Reply:
Yes, you are correct. Actually I forgot to run `time` on `CNN` tests, sorry. I have made the changes.